src/Sentry/Android/SentrySdk.cs

```diff
@@ -1,52 +1,162 @@
+                    //o.DistinctId = ?
+
+                    // TODO: Should we just copy EnableScopeSync, or should we always set it true for Android?
+                    //       And also, Do we need to pass a scope observer?
+                    //o.EnableScopeSync = options.EnableScopeSync; ??
+
+                    // These options are copied over from our SentryOptions
+                    o.AttachStacktrace = options.AttachStacktrace;
+                    o.Debug = options.Debug;
+                    o.DiagnosticLevel = options.DiagnosticLevel.ToJavaSentryLevel();
                     o.Dsn = options.Dsn;
-                    //TODO: we'll need to pass other stuff here from SentryOptions to SentryAndroidOptions
+                    o.EnableAutoSessionTracking = options.AutoSessionTracking;
+                    o.Environment = options.Environment;
+                    o.FlushTimeoutMillis = (long)options.InitCacheFlushTimeout.TotalMilliseconds;
+                    o.MaxAttachmentSize = options.MaxAttachmentSize;
+                    o.MaxBreadcrumbs = options.MaxBreadcrumbs;
+                    o.MaxCacheItems = options.MaxCacheItems;
+                    o.MaxQueueSize = options.MaxQueueItems;
+                    o.Release = options.Release;
+                    o.SampleRate = (JavaDouble?)options.SampleRate;
+                    o.SendClientReports = options.SendClientReports;
+                    o.SendDefaultPii = options.SendDefaultPii;
+                    o.ServerName = options.ServerName;
+                    o.SessionTrackingIntervalMillis = (long)options.AutoSessionTrackingInterval.TotalMilliseconds;
+                    o.ShutdownTimeoutMillis = (long)options.ShutdownTimeout.TotalMilliseconds;
+                    o.TracesSampleRate = (JavaDouble?)options.TracesSampleRate;
+
+                    if (options.CacheDirectoryPath != null)
+                    {
+                        // Set a separate cache path for the native SDK so we don't step on the managed one
+                        o.CacheDirPath = Path.Combine(options.CacheDirectoryPath, "native");
+                    }
+
+                    foreach (var tag in options.DefaultTags)
+                    {
+                        o.SetTag(tag.Key, tag.Value);
+                    }
+
+                    if (options.BeforeBreadcrumb != null)
+                    {
+                        o.BeforeBreadcrumb = new BeforeBreadcrumbCallback(options.BeforeBreadcrumb, options.DiagnosticLogger, o);
+                    }
+
+                    if (options.BeforeSend != null)
+                    {
+                        o.BeforeSend = new BeforeSendCallback(options.BeforeSend, options.DiagnosticLogger, o);
+                    }
+
+                    if (options.TracesSampler != null)
+                    {
+                        o.TracesSampler = new TracesSamplerCallback(options.TracesSampler);
```

Comment:
We often check for the existence of an instance here to make decisions (performance is enabled, add auto instrumentation) so this could have side effects

Reply:
I'll address this in a future PR.  Thanks.